plugin.rb

```diff
@@ -127,4 +127,33 @@ def retorts
       return ActiveModel::ArraySerializer.new(Retort::Retort.for_post(post: object), each_serializer: ::Retort::RetortSerializer).as_json
     end
   end
+
+  require_dependency 'rate_limiter'
+  require_dependency 'post_detail'
+  class ::PostDetail
+    include RateLimiter::OnCreateRecord
+    rate_limit :retort_rate_limiter
+    after_update { run_callbacks :create if is_retort? }
+
```

Comment:
Should these methods be private?

Reply:
Perhaps they could be. The Discourse convention is to not make rate limit methods private.